### ISSUE TITLE
minOccurs="0" maxOccurs="0" not working as expected - document passing validation when it should not

### ISSUE THREAD

**Issue by Zippy1999** [sissaschool/xmlschema#306]
I believe there is a problem validating minOccurs and maxOccurs when both are set to zero.

Sample xsd - _Note that in this example, for element "name", minOccurs=0 and maxOccurs=0_

```
<?xml version="1.0" encoding="UTF-8" ?>
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
    <xs:element name="Document">
        <xs:complexType>
            <xs:sequence>
                <xs:element name="shipto">
                    <xs:complexType>
                        <xs:sequence>
                            <xs:element minOccurs="0" maxOccurs="0" name="name" type="xs:string"/>
                            <xs:element minOccurs="0" maxOccurs="1" name="address" type="xs:string"/>
                        </xs:sequence>
                    </xs:complexType>
                </xs:element>
            </xs:sequence>
        </xs:complexType>
    </xs:element>
</xs:schema>
```

Sample XML file to validate

```
<Document>
  <shipto>
    <name>Bob</name>
    <address>Bob</address>
  </shipto>
</Document>
```

When I try to validate the xml against the xsd, I should expect the xml to fail validation as `name` would be invalid, however both `is_valid` and `validate` pass without issue

**Comment by Zippy1999** [comment]
The example above has been simplified to reproduce the issue. I know that in the real world it may seem odd to see  minOccurs=0 and maxOccurs=0 specified on an element. Usually you would expect the element to be omitted altogether.

To add some extra context, the xsd file in question which has highlighted this issue has a base definition:

```
<xs:complexType name="ReportEntry2">
  <xs:sequence>
    <xs:element name="NtryRef" type="Max35Text" minOccurs="0"/>
    <xs:element name="Amt" type="ActiveOrHistoricCurrencyAndAmount"/>
```

This would then be extended:

```
<xs:complexType name="ReportEntry2_CH_camt052">
  <xs:complexContent>
    <xs:restriction base="ReportEntry2">
      <xs:sequence>
        <xs:element name="NtryRef" type="Max35Text" minOccurs="0" maxOccurs="0"/>
        <xs:element name="Amt" type="ActiveOrHistoricCurrencyAndAmount_CH_camt052_2">
```

Note that NtryRef in the extended section has maxOccurs=0 so an xml file with information like this:

```
                <Ntry>
                    <NtryRef>Some info</NtryRef>
                    <Amt Ccy="GBP">100.00</Amt>
```

should fail validation but it does not.

**Comment by brunato** [comment]
Hi,
surely this is important, despite it should not be used in real schemas, because it puts in evidence bugs in the model validation.
I will try to fix in `ModelVisitor` class if it's possible or put a workarount in element matching.

Thank you

**Comment by Zippy1999** [comment]
Thank you for confirming and looking into this :+1:

**Comment by priyatamnano** [comment]
Also, validation failed if xm attribute value contains  "&"

@brunato

**Comment by brunato** [comment]
Fixed with release v1.11.3. 

The better resolution for this has been adding a `max_occurs` check in `XsdGroup.iter_decode()`. The empty XSD element is simply ignored and the model validation contuinue with an `advance(False)`.

Thank you
